server/sonar-server/src/main/java/org/sonar/server/measure/live/IssueCounter.java

```diff
@@ -0,0 +1,153 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.measure.live;
+
+import java.util.Collection;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.sonar.api.rule.Severity;
+import org.sonar.api.rules.RuleType;
+import org.sonar.db.issue.IssueGroupDto;
+import org.sonar.db.rule.SeverityUtil;
+
+class IssueCounter {
+
+  private final Map<RuleType, HighestSeverity> highestSeverityOfUnresolved = new EnumMap<>(RuleType.class);
+  private final Map<RuleType, Effort> effortOfUnresolved = new EnumMap<>(RuleType.class);
+  private final Map<String, Count> unresolvedBySeverity = new HashMap<>();
+  private final Map<RuleType, Count> unresolvedByType = new EnumMap<>(RuleType.class);
+  private final Map<String, Count> byResolution = new HashMap<>();
+  private final Map<String, Count> byStatus = new HashMap<>();
+  private final Count unresolved = new Count();
+
+  IssueCounter(Collection<IssueGroupDto> groups) {
+    for (IssueGroupDto group : groups) {
+      RuleType ruleType = RuleType.valueOf(group.getRuleType());
+      if (group.getResolution() == null) {
+        highestSeverityOfUnresolved
+          .computeIfAbsent(ruleType, k -> new HighestSeverity())
+          .add(group);
+        effortOfUnresolved
+          .computeIfAbsent(ruleType, k -> new Effort())
+          .add(group);
+        unresolvedBySeverity
+          .computeIfAbsent(group.getSeverity(), k -> new Count())
+          .add(group);
+        unresolvedByType
+          .computeIfAbsent(ruleType, k -> new Count())
+          .add(group);
+        unresolved.add(group);
+      } else {
+        byResolution
+          .computeIfAbsent(group.getResolution(), k -> new Count())
+          .add(group);
+      }
+      if (group.getStatus() != null) {
+        byStatus
+          .computeIfAbsent(group.getStatus(), k -> new Count())
+          .add(group);
+      }
+    }
+  }
+
+  public Optional<String> getHighestSeverityOfUnresolved(RuleType ruleType, boolean onlyInLeak) {
+    return Optional.ofNullable(highestSeverityOfUnresolved.get(ruleType))
+      .map(hs -> hs.severity(onlyInLeak));
+  }
+
+  public double sumEffortOfUnresolved(RuleType type, boolean onlyInLeak) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "onlyInLeak" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2301)